swarm/storage/mru/request.go

```diff
@@ -109,127 +80,80 @@ func NewCreateRequest(name string, frequency, startTime uint64, ownerAddr common
 		},
 	}
 
-	updateRequest.rootAddr, updateRequest.metaHash, _ = updateRequest.resourceMetadata.hash()
+	request.rootAddr, request.metaHash, _ = request.resourceMetadata.hash()
 
-	return updateRequest, nil
+	return request, nil
 }
 
 // Frequency Returns the resource expected update frequency
-func (r *UpdateRequest) Frequency() uint64 {
+func (r *Request) Frequency() uint64 {
 	return r.frequency
 }
 
 // Name returns the resource human-readable name
-func (r *UpdateRequest) Name() string {
+func (r *Request) Name() string {
 	return r.name
 }
 
 // Multihash returns true if the resource data should be interpreted as a multihash
-func (r *UpdateRequest) Multihash() bool {
+func (r *Request) Multihash() bool {
 	return r.multihash
 }
 
+// Period returns in which period the resource will be published
+func (r *Request) Period() uint32 {
+	return r.period
+}
+
 // Version returns the resource version to publish
-func (r *UpdateRequest) Version() uint32 {
+func (r *Request) Version() uint32 {
 	return r.version
 }
 
-// Period returns in which period the resource will be published
-func (r *UpdateRequest) Period() uint32 {
-	return r.period
+// RootAddr returns the metadata chunk address
```

Comment:
There are three Addresses in a request:
* `OwnerAddr` the resource owner's ethereum address
* `RootAddr` the root metadata chunk storage address
* `UpdateAddr`: the update lookup chunk address, recently renamed to just `Addr` from `GetUpdateAddr` and looking at this I just renamed it to `UpdateAddr` for consistency.

The `Request` struct can have the three, because a `Request` combines metadata+update, and `Addr()` would collide. 

I had to refactor code already where everything was "addr" or "key" and was very confusing to follow. Therefore I'd like to have a very clear distinction going forward.

Also, this struct is serialized to JSON and we lose type information, so I'd like the JSON user to know it is a `0x` address they have to put in that string.

I can rename `OwnerAddr` to just `Owner`, but in JSON it will be `ownerAddr`.